src/redux/tablesRedux.js

```diff
@@ -0,0 +1,108 @@
+// import { StarsOutlined } from '@material-ui/icons';
+import Axios from 'axios';
+import { api } from '../settings';
+
+/* selectors */
+export const getAll = ({ tables }) => tables.data;
+export const getLoadingState = ({ tables }) => tables.loading;
+// export const getTables = ({}) => tables;
+
+/* action name creator */
+const reducerName = 'tables';
+const createActionName = (name) => `app/${reducerName}/${name}`;
+
+/* action types */
+const FETCH_START = createActionName('FETCH_START');
+const FETCH_SUCCESS = createActionName('FETCH_SUCCESS');
+const FETCH_ERROR = createActionName('FETCH_ERROR');
+const FETCH_STATUS = createActionName('FETCH_STATUS');
+
+/* action creators */
+export const fetchStarted = (payload) => ({ payload, type: FETCH_START });
+export const fetchSuccess = (payload) => ({ payload, type: FETCH_SUCCESS });
+export const fetchError = (payload) => ({ payload, type: FETCH_ERROR });
+export const changeTableStatus = (status, id) => ({
+  payload: { status: status, id: id },
+  type: FETCH_STATUS,
+});
+
+/* thunk creators */
+export const fetchFromAPI = () => {
+  return (dispatch, getState) => {
+    dispatch(fetchStarted());
+
+    Axios.get(`${api.url}/api/${api.tables}`)
+      .then((res) => {
+        dispatch(fetchSuccess(res.data));
+      })
+      .catch((err) => {
+        dispatch(fetchError(err.message || true));
+      });
+  };
+};
+
+export const changeTableStatusInAPI = (status, id) => {
+  return (dispatch, getState) => {
+    dispatch(fetchStarted());
+
+    Axios.post(`${api.url}/api/${api.tables}`)
+      .then((res) => {
+        dispatch(changeTableStatus(status, id));
+      })
+      .catch((err) => {
+        dispatch(fetchError(err.message || true));
+      });
+  };
+};
+
+/* reducer */
+export default function reducer(statePart = [], action = {}) {
```

Comment:
`statePart` nie jest tablicą!! to powinien byc obiekt z wartościami początkowymi takimi jak w pliku `store`
Sugeruje zrobić stałą `initialTableState` i ją wyeskportować 

```js
export const initialTableState = {
    data: [],
    loading: {
      active: false,
      error: false,
    },
  }
```
```suggestion
export default function reducer(statePart = initialTableState, action) {
```
Dla `action` nie jest potrzeban wartośc domyślna. switch i tak wyłapie case jeśli nie zostanie przekazana czyli będzie miała wartośc `undefined`

Reply:
Done.